### ISSUE TITLE
Hide schema editing menu and tabs

### ISSUE THREAD

**Issue by marcpley** [pawelsalawa/sqlitestudio#5159]
In order to have a data only mode, add an option to hide menu Structure and tab selection.

![image](https://github.com/user-attachments/assets/98c61846-dcca-430f-8007-575507ba92e7)
Add checkbox "Hide schema editing menu and tabs"

**Comment by marcpley** [comment]
The Structure menu could be visible and inactive. I think it would be more comprehensible for the user.
The tab selector could be not visible instead of sowing a single Data tab. This spare vertical room for the data.

**Comment by pawelsalawa** [comment]
I think all settings for the "data only mode" that you're aiming here should be placed on a dedicated page tab in Configuration Dialog, like - for example - `ScriptingPython` plugin does in 3.5.x:
![image](https://github.com/user-attachments/assets/16e4f81b-b7b4-43a5-bd46-503faa93673a)
...or similarly does the "SQL enterprise" formatter plugin.

Since plugin will implement/handle all these settings, they should be on its dedicated page.

As for the functionality - I guess hiding the structure menu (and structure toolbar too) is fine. Keeping it visible if it's intended to be disabled all the time does not make much sense - it only keeps the UI more "noisy" in that case.

Hiding table tabs other than "Data" is possible, but it will require to connect to signals of MDI area instance, to handle events of opening `TableWindow` and hiding the tabs.

**Comment by marcpley** [comment]
Does the export/import settings will deal with plugins params too ?

**Comment by marcpley** [comment]
Arghhh

**Comment by pawelsalawa** [comment]
?

**Comment by pawelsalawa** [comment]
I meant #5106 will be part of core functionality. Did you understand something else?

**Comment by pawelsalawa** [comment]
Settings import/export is more for cases when people (for various unknown reasons) are unable to use the application anymore. For example it happens sometimes that someone upgrades SQLiteStudio version and suddenly it no longer works, but it's fine if they rename settings file. This means some particular setting is causing crash for them. In that case it makes sense to be able to import Functions, Collations, Database List - from old configuration, while skipping the core settings (style, fonts, colors, etc), which is most likely responsible for the crash.

For your needs I would recommend creating `sqlitestudio-cfg` directory next to the SQLiteStudio.exe - in that case SQLiteStudio will store its configuration there. Then you can just distribute this directory alongside of the SQLiteStudio itself and if someone runs it with your `sqlitestidio-cfg` located next to their SQLiteStudio.exe - they will use your copy of settings.

**Comment by pawelsalawa** [comment]
This `sqlitestudio-cfg` is described at https://github.com/pawelsalawa/sqlitestudio/wiki/Configuration_directory_location (at the bottom).

**Comment by marcpley** [comment]
Does the plugins params are in sqlitestudio-cfg too ?

**Comment by marcpley** [comment]
I create the dir /opt/SQLiteStudio/sqlitestudio-cfg (linux) but it stay empty afer running and closing SQLiteStudio...
How this dir is more portable than settings3 file ?

**Comment by pawelsalawa** [comment]
Is your SQLiteStudio executable that you run `/opt/SQLiteStudio/sqlitestudio`?

**Comment by pawelsalawa** [comment]
(and yes, plugins settings/params will be stored in that file too)

**Comment by marcpley** [comment]
Yes I run /opt/SQLiteStudio/sqlitestudio

**Comment by pawelsalawa** [comment]
Is `/opt/SQLiteStudio/sqlitestudio-cfg` writable by the user that runs SQLiteStudio?

**Comment by pawelsalawa** [comment]
`/home/marc/.config/SalSoft/SQLiteStudio` will not be treated as portable.

The `sqlitestudio-cfg` can be placed either next to the executable binary, or (and this is a chance for you) in the folder from which you start SQLiteStudio. If you start it from terminal - you will always know where is your current directory. If you start from "menu start icon", then probably the working directory is your home directory.

Summing up - I recommend you to create `/home/marc/sqlitestudio-cfg` and see if that works.

**Comment by marcpley** [comment]
It works with /home/marc/sqlitestudio-cfg
Creating a dir here is a bit excessive!  But  the folder from which start SQLiteStudio opens possibilities.
Anyway, if I create a specific app, this is less important.
